cfgmgr/buffermgr.cpp

```diff
@@ -442,7 +458,8 @@ void BufferMgr::doTask(Consumer &consumer)
                 if (m_speedLookup.count(port) != 0)
                 {
                     // create/update profile for port
-                    task_status = doSpeedUpdateTask(port, admin_up);
+                    string lossless_pgs = getPfcEnableQueuesForPort(port);
```

Comment:
This is still risky. It assumes when BUFFER_PG is created all `PORT`, `CABLE_LENGTH` and `PORT_QOS_MAP` have been received, which is not guaranteed.
Consider this scenario:
1. buffer manager receives the `PORT` table and the `CABLE_LENGTH` table. It will handle them and generate `BUFFER_PG` table. But it hasn't received `PORT_QOS_MAP` yet, so the lossless PGs will be `3-4`.
2. Later it receives `PORT_QOS_MAP` table which contains `2,3-4,6` for dual ToR ports. but corresponding `BUFFER_PG` will not be updated.

I suggest generating `BUFFER_PG` only after all tables are available and not using `LOSSLESS_PGS` in case `pfc_enable` is empty. (I have a vague impression that removing BUFFER_PG is not supported on some platform.)

Reply:
@stephenxs Thanks for the good catch. I update the code to add a retry if `PORT_QOS_MAP` table is not available. Please help take a look once more.

Reply:
Looks like there is still an issue in case a port's `pfc_enable` changed. The lossless buffers for the priorities that were but aren't enabled are kept untouched. In fact, they should be removed.
Probably we need to
- save `pfc_enable` for each port
- when `pfc_enable` is updated, remove the `BUFFER_PG` items generated according to the previous ones before generating the new ones.

sorry for the late response

Reply:
As discussed offline, we will not support update `BUFFER_PG` without a fully reload. So the change is reverted.